src/main/java/com/google/codeu/data/UserDatastore.java

```diff
@@ -99,5 +100,19 @@ public User getUser(String id) {
 	    users.add((String) entity.getProperty("email"));
 	  }
 	  return users;
-	}
+  }
+  
+  public List<User> getUsersForTeam(String teamId) {
+	  List<User> users = new ArrayList<User>();
+	  Query query = new Query("User")
+	  .setFilter(new Query.FilterPredicate("teamId", FilterOperator.EQUAL, teamId));
+	  PreparedQuery results = userdatastore.prepare(query);
+	   List<Entity> userEntities = results.asList(FetchOptions.Builder.withLimit(4));
+	   if (userEntities == null) {
+	    return null;
+	   }
+	   for (Entity userEntity : userEntities) 
+		   users.add(getUser((String) userEntity.getProperty("id")));
```

Comment:
This does work but we are reading the database twice for the same bit of information. When getUser is called here we are querying the database for  information that is already available in the enity. I'd recommend extracting some of the following code in getUser() into a method that returns a User model given an entity. Like so:

```java
public User getUserFromEntity(Entity entity) {
  String email = userEntity.getKey().getName();
   String aboutMe = (String) userEntity.getProperty("aboutMe");
   String imgUrl = (String) userEntity.getProperty("imgUrl");
   String firstName = (String) userEntity.getProperty("firstName");
   String lastName = (String) userEntity.getProperty("lastName");
   String teamId = (String) userEntity.getProperty("teamId");
   return new User(UUID.fromString(id), email, teamId, aboutMe, firstName, lastName, imgUrl);
}
```